File: internal/model/weather.go
package model

import "time"

// WeatherRequest 天气查询请求结构体
type WeatherRequest struct {
	City      string  `json:"city" form:"city" binding:"required_without=Lat"`           // 城市名称
	Lat       float64 `json:"lat" form:"lat" binding:"required_without=City"`            // 纬度
	Lon       float64 `json:"lon" form:"lon" binding:"required_with=Lat"`                // 经度
	Units     string  `json:"units" form:"units" binding:"omitempty,oneof=metric imperial standard"` // 单位系统
	Lang      string  `json:"lang" form:"lang" binding:"omitempty"`                      // 语言
}

// WeatherResponse 标准化的天气响应结构体
type WeatherResponse struct {
	Location    Location    `json:"location"`    // 位置信息
	Current     Current     `json:"current"`     // 当前天气
	Timestamp   int64       `json:"timestamp"`   // 响应时间戳
	Provider    string      `json:"provider"`    // 数据提供商
}

// Location 位置信息
type Location struct {
	Name      string  `json:"name"`      // 城市名称
	Country   string  `json:"country"`   // 国家代码
	Latitude  float64 `json:"latitude"`  // 纬度
	Longitude float64 `json:"longitude"` // 经度
	Timezone  int     `json:"timezone"`  // 时区偏移（秒）
}

// Current 当前天气信息
type Current struct {
	Temperature     float64     `json:"temperature"`      // 当前温度
	FeelsLike       float64     `json:"feels_like"`       // 体感温度
	TempMin         float64     `json:"temp_min"`         // 最低温度
	TempMax         float64     `json:"temp_max"`         // 最高温度
	Pressure        int         `json:"pressure"`         // 大气压力（hPa）
	Humidity        int         `json:"humidity"`         // 湿度（%）
	Visibility      int         `json:"visibility"`       // 能见度（米）
	UVIndex         float64     `json:"uv_index"`         // 紫外线指数
	Weather         []Weather   `json:"weather"`          // 天气状况
	Wind            Wind        `json:"wind"`             // 风力信息
	Clouds          Clouds      `json:"clouds"`           // 云量信息
	Rain            *Rain       `json:"rain,omitempty"`   // 降雨信息
	Snow            *Snow       `json:"snow,omitempty"`   // 降雪信息
	Sunrise         int64       `json:"sunrise"`          // 日出时间戳
	Sunset          int64       `json:"sunset"`           // 日落时间戳
	UpdatedAt       time.Time   `json:"updated_at"`       // 数据更新时间
}

// Weather 天气状况
type Weather struct {
	ID          int    `json:"id"`          // 天气状况 ID
	Main        string `json:"main"`        // 天气主要状况
	Description string `json:"description"` // 天气详细描述
	Icon        string `json:"icon"`        // 天气图标代码
}

// Wind 风力信息
type Wind struct {
	Speed     float64 `json:"speed"`     // 风速
	Direction int     `json:"direction"` // 风向（度）
	Gust      float64 `json:"gust"`      // 阵风速度
}

// Clouds 云量信息
type Clouds struct {
	All int `json:"all"` // 云量百分比
}

// Rain 降雨信息
type Rain struct {
	OneHour   float64 `json:"1h,omitempty"`   // 过去1小时降雨量（mm）
	ThreeHour float64 `json:"3h,omitempty"`   // 过去3小时降雨量（mm）
}

// Snow 降雪信息
type Snow struct {
	OneHour   float64 `json:"1h,omitempty"`   // 过去1小时降雪量（mm）
	ThreeHour float64 `json:"3h,omitempty"`   // 过去3小时降雪量（mm）
}

// ErrorResponse 错误响应结构体
type ErrorResponse struct {
	Error   string `json:"error"`             // 错误信息
	Code    int    `json:"code"`              // 错误代码
	Message string `json:"message,omitempty"` // 详细错误信息
}

// APIResponse 通用 API 响应结构体
type APIResponse struct {
	Success bool        `json:"success"`           // 请求是否成功
	Data    interface{} `json:"data,omitempty"`    // 响应数据
	Error   *ErrorResponse `json:"error,omitempty"` // 错误信息
}

File: frontend/src/types/weather.ts
// 天气数据类型定义

export interface WeatherRequest {
  city?: string
  lat?: number
  lon?: number
  units?: 'metric' | 'imperial' | 'standard'
  lang?: string
}

export interface Location {
  name: string
  country: string
  latitude: number
  longitude: number
  timezone: number
}

export interface Weather {
  id: number
  main: string
  description: string
  icon: string
}

export interface Wind {
  speed: number
  direction: number
  gust: number
}

export interface Clouds {
  all: number
}

export interface Rain {
  '1h'?: number
  '3h'?: number
}

export interface Snow {
  '1h'?: number
  '3h'?: number
}

export interface Current {
  temperature: number
  feels_like: number
  temp_min: number
  temp_max: number
  pressure: number
  humidity: number
  visibility: number
  uv_index: number
  weather: Weather[]
  wind: Wind
  clouds: Clouds
  rain?: Rain
  snow?: Snow
  sunrise: number
  sunset: number
  updated_at: string
}

export interface WeatherResponse {
  location: Location
  current: Current
  timestamp: number
  provider: string
}

export interface APIResponse<T = any> {
  success: boolean
  data?: T
  error?: {
    error: string
    code: number
    message?: string
  }
}

// 地理位置相关类型
export interface GeolocationPosition {
  coords: {
    latitude: number
    longitude: number
    accuracy: number
  }
}

export interface GeolocationError {
  code: number
  message: string
}

// UI 状态类型
export interface LoadingState {
  isLoading: boolean
  error: string | null
}

// 天气图标映射
export const WEATHER_ICONS: Record<string, string> = {
  '01d': 'sun',
  '01n': 'moon',
  '02d': 'cloud-sun',
  '02n': 'cloud-moon',
  '03d': 'cloud',
  '03n': 'cloud',
  '04d': 'clouds',
  '04n': 'clouds',
  '09d': 'cloud-rain',
  '09n': 'cloud-rain',
  '10d': 'cloud-sun-rain',
  '10n': 'cloud-moon-rain',
  '11d': 'cloud-lightning',
  '11n': 'cloud-lightning',
  '13d': 'cloud-snow',
  '13n': 'cloud-snow',
  '50d': 'cloud-fog',
  '50n': 'cloud-fog',
}

// 天气背景颜色映射
export const WEATHER_BACKGROUNDS: Record<string, string> = {
  'clear': 'from-blue-400 to-blue-600',
  'clouds': 'from-gray-400 to-gray-600',
  'rain': 'from-gray-500 to-blue-700',
  'drizzle': 'from-gray-400 to-blue-500',
  'thunderstorm': 'from-gray-700 to-purple-900',
  'snow': 'from-gray-300 to-blue-400',
  'mist': 'from-gray-400 to-gray-500',
  'fog': 'from-gray-400 to-gray-500',
  'haze': 'from-yellow-300 to-orange-400',
  'default': 'from-blue-400 to-purple-600',
}
